File: sources/mobile/app/src/main/java/com/console/ratcord/domain/entity/expense/ExpenseMinimal.kt
package com.console.ratcord.domain.entity.expense

import androidx.room.PrimaryKey
import com.console.ratcord.domain.entity.user.UserMinimalWithId
import kotlinx.serialization.Serializable

@Serializable
data class ExpenseMinimal(
    @PrimaryKey val id: Int,
    val user: UserMinimalWithId,
    val groupId: Int,
    val userIdsInvolved: List<Int>,
    val categoryId: Int,
    val amount: Float,
    val date: Int,
    val place: String,
    val description: String,
)

File: sources/api/Models/Expense/ExpenseMinimal.cs
namespace DotNetAPI.Models.Expense
{
    public class ExpenseMinimal
    {
        public int Id { get; set; }
        public required User.UserDTO User { get; set; }
        public required int GroupId { get; set; }
        public required IList<int> UserIdInvolved { get; set; }
        public required int CategoryId { get; set; }
        public required float Amount { get; set; }
        public required int Date { get; set; }
        public required string Place { get; set; }
        public string? Description { get; set; }
    }
}
